TestFoundation/FTPServer.swift

```
@@ -0,0 +1,274 @@
+        #else
+            let SOCKSTREAM = SOCK_STREAM
+        #endif
+        listenSocket = try attempt("socket", valid: isNotNegative, socket(AF_INET, SOCKSTREAM, Int32(IPPROTO_TCP)))
+        var on: Int = 1
+        _ = try attempt("setsockopt", valid: isZero, setsockopt(listenSocket, SOL_SOCKET, SO_REUSEADDR, &on, socklen_t(MemoryLayout<Int>.size)))
+        let sa = createSockaddr(port)
+        socketAddress.initialize(to: sa)
+        try socketAddress.withMemoryRebound(to: sockaddr.self, capacity: MemoryLayout<sockaddr>.size, {
+            let addr = UnsafePointer<sockaddr>($0)
+            _ = try attempt("bind", valid: isZero, bind(listenSocket, addr, socklen_t(MemoryLayout<sockaddr>.size)))
+        })
+        
+        dataSocket = try attempt("socket", valid: isNotNegative, socket(AF_INET, SOCKSTREAM, Int32(IPPROTO_TCP)))
+        var on1: Int = 1
+        _ = try attempt("setsockopt", valid: isZero, setsockopt(dataSocket, SOL_SOCKET, SO_REUSEADDR, &on1, socklen_t(MemoryLayout<Int>.size)))
+        let sa1 = createSockaddr(port+1)
+        socketAddress1.initialize(to: sa1)
+        try socketAddress1.withMemoryRebound(to: sockaddr.self, capacity: MemoryLayout<sockaddr>.size, {
+            let addr = UnsafeMutablePointer<sockaddr>($0)
+            _ = try attempt("bind", valid: isZero, bind(dataSocket, addr, socklen_t(MemoryLayout<sockaddr>.size)))
+            var sockLen = socklen_t(MemoryLayout<sockaddr>.size)
+            _ = try attempt("listen", valid: isZero, listen(dataSocket, SOMAXCONN))
+            // Open the data port asynchronously. Port should be opened before ESPV header communication.
+            DispatchQueue(label: "delay").async {
+                do {
+                    self.dataSocket = try self.attempt("accept", valid: self.isNotNegative, accept(self.dataSocket, addr, &sockLen))
+                    self.dataSocketPort = sa1.sin_port
+                } catch {
+                }
```

I know this is test code, but do you really mean to eat any error and continue? Seems like the test should abort.

---

Isn't the entire purpose of the test suite to log failures?